packages/tpetra/core/test/CrsGraph/CrsGraph_UnitTests_Swap.cpp

```diff
@@ -44,348 +44,434 @@
+    using vec_weighted_edge_type   = std::vector<tuple_weighted_edge_type>;                 // Vector of weighted edges
+
+  public:
+    void execute(Teuchos::FancyOStream& out, bool& success)
+    {
+        using std::endl;
+        using Teuchos::Comm;
+        using Teuchos::outArg;
+        using Teuchos::RCP;
 
-#define STD_TESTS(graph)                                                                                      \
-    {                                                                                                         \
-        auto   STCOMM   = graph.getComm();                                                                    \
-        auto   STMYGIDS = graph.getRowMap()->getNodeElementList();                                            \
-        size_t STMAX    = 0;                                                                                  \
-                                                                                                              \
-        for(size_t STR = 0; STR < graph.getNodeNumRows(); ++STR)                                              \
-        {                                                                                                     \
-            TEST_EQUALITY(graph.getNumEntriesInLocalRow(STR), graph.getNumEntriesInGlobalRow(STMYGIDS[STR])); \
-            STMAX = std::max(STMAX, graph.getNumEntriesInLocalRow(STR));                                      \
-        }                                                                                                     \
-        TEST_EQUALITY(graph.getNodeMaxNumRowEntries(), STMAX);                                                \
-        GST STGMAX;                                                                                           \
-        Teuchos::reduceAll<int, GST>(*STCOMM, Teuchos::REDUCE_MAX, STMAX, Teuchos::outArg(STGMAX));           \
-        TEST_EQUALITY(graph.getGlobalMaxNumRowEntries(), STGMAX);                                             \
-    }
+        bool verbose = Tpetra::Details::Behavior::verbose();
 
+        comm_type initialComm = getDefaultComm();
 
-TEUCHOS_STATIC_SETUP()
-{
-    Teuchos::CommandLineProcessor& clp = Teuchos::UnitTestRepository::getCLP();
-    clp.setOption("filedir", &filedir, "Directory of expected input files.");
-    clp.addOutputSetupOptions(true);
-    clp.setOption("test-mpi",
-                  "test-serial",
-                  &Tpetra::TestingUtilities::testMpi,
-                  "Test MPI (if available) or force test of serial.  In a serial build,"
-                  " this option is ignored and a serial comm is always used.");
-    clp.setOption("error-tol-slack", &errorTolSlack, "Slack off of machine epsilon used to check test results");
-}
+        TEUCHOS_TEST_FOR_EXCEPTION(initialComm->getSize() < 2, std::runtime_error, "This test requires at least two processors.");
```

Comment:
This should really be avoided, throwing on `comm-getSize()` leads to failing tests, you really want you test to handle all cases, after all are users forbidden to call this method on serial matrices?

Reply:
I agree with @lucbv .  swap() should work for serial builds and MPI builds run on one rank, so we should test those cases.  Is the two-rank case just a legacy from an import test or similar?  

Are there any problems we might catch on more than two processors that we wouldn't see on only two?  ("No" is an acceptable answer if you think it is true.)

Reply:
@kddevin @lucbv 
There's nothing in Swap that's tied to any particular configuration of processors, so swap() itself should work fine since we're generally using std::swap() in most places. 

Generating the matrices and graphs for 2 processes is pretty much the result of me being fairly new to generating CrsGraphs and CrsMatrices and there aren't a lot of small(ish) self-contained examples that made much sense to me prior to this which can just generate a simple graph, so I used [this example][1] to generate the graphs, which followed on to the matrix stuff by copy and pasting.

I think it'd be good to rework this to work for 1..N processors... but if we make that a blocker on this PR I can kill it, but it's highly unlikely that I'll be getting that done until mid to late April at a minimum since I have to switch over to other projects for a while.  Can we put in an issue to rework the Swap tests to support 1..N processors so that this can get in for the FE* stuff?

[1]: https://github.com/trilinos/Trilinos/wiki/Tpetra::CrsGraph-creation,-from-a-graph-perspective